### DISCUSSION TITLE
graphql/scalars/docs/quick-start

### DISCUSSION THREAD

**Discussion by giscus** [Urigo/graphql-scalars#2124] [Docs Discussions]
# graphql/scalars/docs/quick-start

GraphQL Scalars Documentation

https://the-guild.dev/graphql/scalars/docs/quick-start

**Comment by ankit9787** [comment]
Hey do you have any code or sample code repo which we can run and see how custom scalar works ?? because theortically hard to understand its usage

**Reply by ardatan** [reply]
You can refer to this documentation;
https://the-guild.dev/graphql/tools/docs/scalars

**Comment by djMax** [comment]
The codegen integration sample isn't great - can you explain why those three (and how to add more), and how the string type should be set, if at all?

**Reply by user928** [reply]
I tried this but it is not working for NonEmptyString

import { NonEmptyStringResolver } from 'graphql-scalars';

and then in CodegenConfig

config: {
				scalars: {
					NonEmptyString: NonEmptyStringResolver.extensions.codegenScalarType,
				},
			},

**Reply by ardatan** [reply]
PRs are welcome to improve the example. Those three are just for example. You can add more for sure.
For `NonEmptyString`, you can just use `string`.

**Reply by user928** [reply]
Thanks, but I was thinking that custom scalars like NonEmptyString will validate in runtime and prevent me from using empty string in my code, if I set it to just string, I can still use empty string in code without any ts warning

**Reply by jetaggart** [reply]
How are we supposed to use this with clientside codegen for something like a `scalar Time` which is the DateTime scalar in the libary?
